suzieq/db/parquet/parquetdb.py

```diff
@@ -509,16 +509,8 @@ def _get_cp_dataset(self, table_name: str, need_sqvers: bool,
 
         folder = self._get_table_directory(table_name, True)
 
-        # pylint: disable=simplifiable-if-statement
-        if not (start_time or end_time) and (view == "all"):
-            # Enforcing the logic we have: if both start_time & end_time
-            # are given, return all files since the model is that the user is
-            # expecting to see all changes in the time window. Otherwise, user
-            # expecting to see only the latest before an end_time OR after a
-            # start_time.
-            all_files = True
-        else:
-            all_files = False
+        # pylint: disable=consider-using-ternary
+        all_files = ((start_time and end_time) or (view == "all"))
```

Comment:
all_files is true only for both start_time and end_time are set (or view == 'all'). Shouldn't we set all_files to False also if only one between start_time or end_time are specified?
Isn't it `all_files = (not (start_time or end_time) or (view == 'all'))` what we should do?

Reply:
I agree with Luca, with this change if we set both `start_time` and `end_time` we are going to read every file of the dataset.

Reply:
I agree, Ugh. I had the original code, but it'd read all files if we only gave end-time. I don't the logic is here that needs to be fixed.

Reply:
The correct logic I think is this in English; if view = all is specified, and there are no time parameters, just suck in all files. So, its (view == "all") and not (start_time or end_time)